cmd/placement/main.go

```diff
@@ -16,100 +15,99 @@ import (
 	"github.com/dapr/dapr/pkg/fswatcher"
 	"github.com/dapr/dapr/pkg/health"
 	"github.com/dapr/dapr/pkg/logger"
-	"github.com/dapr/dapr/pkg/metrics"
 	"github.com/dapr/dapr/pkg/placement"
 	"github.com/dapr/dapr/pkg/placement/monitoring"
+	"github.com/dapr/dapr/pkg/placement/raft"
 	"github.com/dapr/dapr/pkg/version"
 )
 
 var log = logger.NewLogger("dapr.placement")
-var certChainPath string
-var tlsEnabled bool
-
-const (
-	defaultCredentialsPath = "/var/run/dapr/credentials"
-	defaultHealthzPort     = 8080
-	defaultPlacementPort   = 50005
-)
 
 func main() {
-	placementPort := flag.Int("port", defaultPlacementPort, "sets the gRPC port for the placement service")
-	healthzPort := flag.Int("healthz-port", defaultHealthzPort, "sets the HTTP port for the healthz server")
-
-	loggerOptions := logger.DefaultOptions()
-	loggerOptions.AttachCmdFlags(flag.StringVar, flag.BoolVar)
-
-	metricsExporter := metrics.NewExporter(metrics.DefaultMetricNamespace)
-	metricsExporter.Options().AttachCmdFlags(flag.StringVar, flag.BoolVar)
+	log.Infof("starting Dapr Placement Service -- version %s -- commit %s", version.Version(), version.Commit())
 
-	flag.StringVar(&certChainPath, "certchain", defaultCredentialsPath, "Path to the credentials directory holding the cert chain")
-	flag.BoolVar(&tlsEnabled, "tls-enabled", false, "Should TLS be enabled for the placement gRPC server")
-	flag.Parse()
+	cfg := newConfig()
 
-	// Apply options to all loggers
-	if err := logger.ApplyOptionsToLoggers(&loggerOptions); err != nil {
+	// Apply options to all loggers.
+	if err := logger.ApplyOptionsToLoggers(&cfg.loggerOptions); err != nil {
 		log.Fatal(err)
 	}
+	log.Infof("log level set to: %s", cfg.loggerOptions.OutputLevel)
 
-	log.Infof("starting Dapr Placement Service -- version %s -- commit %s", version.Version(), version.Commit())
-	log.Infof("log level set to: %s", loggerOptions.OutputLevel)
-
-	// Initialize dapr metrics exporter
-	if err := metricsExporter.Init(); err != nil {
+	// Initialize dapr metrics for placement.
+	if err := cfg.metricsExporter.Init(); err != nil {
 		log.Fatal(err)
 	}
 
 	if err := monitoring.InitMetrics(); err != nil {
 		log.Fatal(err)
 	}
 
-	stop := make(chan os.Signal, 1)
-	signal.Notify(stop, os.Interrupt)
+	// Start Raft cluster.
+	raftServer := raft.New(cfg.raftID, cfg.raftInMemEnabled, cfg.raftBootStrap, cfg.raftPeers)
```

Comment:
Will this implementation be always using raft or we plan to have a feature flag?

Reply:
placement will always use raft. for single instance mode, it will use inmemory state.